controllers/workspace/devworkspace_controller.go

```diff
@@ -299,6 +300,8 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	devfilePodAdditions, err := containerlib.GetKubeContainersFromDevfile(
+		r.NonCachingClient,
+		workspace.Namespace,
```

Comment:
I think using the `NonCachingClient` for this purpose is a problem -- the purpose of the main controller's client is that it makes far fewer API requests (basically, reads are free due to watches/caching, and only writes require separate API calls). This approach will lead to two API requests per reconcile, regardless of whether the controller does any work, and I worry that this will impact stability.